Create Sui DevNet guide under Explore (#1723)
MystenLabs/sui: doc/src/explore/devnet.md

---
title: Experiment with Sui DevNet
---

Welcome to the beginnings of the Sui DevNet. It exists now to gain operational experience with the Sui software in a public setting. The Sui DevNet currently consists of:

* A four-validator network with all nodes operated by Mysten Labs. Clients send transactions and read requests via this endpoint: http://gateway.devnet.sui.io/
* A public network [Sui Explorer](https://github.com/MystenLabs/sui/tree/main/explorer/client#readme) for browsing the TestNet transaction history: http://explorer.devnet.sui.io
* A [Discord channel](https://discordapp.com/channels/916379725201563759/971488439931392130) for requesting test coins that can be used to pay for gas on the test network. These coins have no financial value and will disappear each time we reset the network.

Many improvements to the Sui DevNet are underway, such as the ability to run full nodes and use a browser-based wallet. See the [Sui DevNet blog post](https://medium.com/mysten-labs) announcement for full details on upcoming features.

> ## Components
> 
> DevNet deployment features the following components of the Sui blockchain:

Reviewer: ```suggestion
## Tools

DevNet provides the following tools for users to interact with the Sui network:
```

Technically we don't deploy "wallet CLI" and they are not components of the blockchain. They are tools to interact with the blockchain

Remember, you can confirm the existence of a command in your PATH by running `which` followed by the command, for example:

```shell
$ which wallet
```
You should see the path to the command. Otherwise, reinstall.

> **Tip:** To reliably test DevNet with the latest Sui binaries, re-install them at least weekly.

In addition, to conduct advanced work such as publishing a Move module or making a Move call, also obtain:

1. a [GitHub account](https://github.com/signup) if you don't have one already
1. the [Sui source code](../build/install.md#source-code); for simplicity, we recommend installing in `~/sui` or using an environment variable

### Set up wallet, connect to gateway

Now [set up your wallet and connect to DevNet](../build/wallet.md#connect-to-devnet) in a single step. Note you can [manually change the Gateway URL](../build/wallet.md#manually-changing-the-gateway-url) if you have already configured a Sui wallet.

> **Tip:** If you run into issues, reset the Sui configuration by removing its directory, by default located at `~/.sui/sui_config`. Then reinstall [Sui binaries](../build/install.md#binaries).

## Basic testing

### Request gas tokens

> Request gas tokens in the [#request-devnet-test-coins](https://discordapp.com/channels/916379725201563759/971488439931392130) Discord channel if you haven't already and wait for them to be issued.

Reviewer: We probably need to add something like the following to this paragraph @sblackshear 

You can request gas tokens by posting your wallet address in the the [#request-devnet-test-coins](https://discordapp.com/channels/916379725201563759/971488439931392130) Discord channel.

You can find your wallet address through the following command

```
$ wallet active-address
F16A5AEDCDF9F2A9C2BD0F077279EC3D5FF0DFEE
```

Author: Let's add that when you all figure it out. Proceeding with what we know now.

Author: Done

---
title: Experiment with Sui DevNet
---

Welcome to the beginnings of the Sui DevNet. It exists now to gain operational experience with the Sui software in a public setting. The Sui DevNet currently consists of:

* A four-validator network with all nodes operated by Mysten Labs. Clients send transactions and read requests via this endpoint: http://gateway.devnet.sui.io/
* A public network [Sui Explorer](https://github.com/MystenLabs/sui/tree/main/explorer/client#readme) for browsing the TestNet transaction history: http://explorer.devnet.sui.io

> * A [Discord channel](https://discordapp.com/channels/916379725201563759/971488439931392130) for requesting test coins that can be used to pay for gas on the test network. These coins have no financial value and will disappear each time we reset the network.

Reviewer: meaning we just add a wallet cli bullet point here and do not mention "faucet" since it's not accessible to external users.

Author: We have other references to faucet here. How should we handle them?

Reviewer: Where are the other references? I tried to search "faucet" but did not find any.

Author: > Where are the other references? I tried to search "faucet" but did not find any.

Disregard. That was in the draft.

Many improvements to the Sui DevNet are underway, such as the ability to run full nodes and use a browser-based wallet. See the [Sui DevNet blog post](https://medium.com/mysten-labs) announcement for full details on upcoming features.

## Purpose

DevNet deployment features the following components of the Sui blockchain:

> 

Reviewer: Feels a bit weird that this section is called `Purpose`. I think it would be better if we just delete this section, and add them to the previous section

Author: Agreed. Replaced with Components. Just trying to split up these lists. Think modular.

    - View transactions and objects

## Prerequisites

### Set up environment

To use the Sui DevNet, you will need:

1. Sui test coins (tokens) requested through [Discord](https://discordapp.com/channels/916379725201563759/971488439931392130)
1. A command line terminal, as virtually everything done here is done by CLI
1. the [`git` command line interface](https://git-scm.com/download/)
1. The [Rust and Cargo](https://doc.rust-lang.org/cargo/getting-started/installation.html) toolchain, updated with `rustup update stable`
1. [Sui binaries](../build/install.md#binaries) in your PATH environment variable, particularly `wallet`

Remember, you can confirm the existence of a command in your PATH by running `which` followed by the command, for example:

```shell
$ which wallet
```
You should see the path to the command. Otherwise, reinstall.

> **Tip:** To reliably test DevNet with the latest Sui binaries, re-install them at least weekly.

In addition, to conduct advanced work such as publishing a Move module or making a Move call, also obtain:

1. a [GitHub account](https://github.com/signup) if you don't have one already

> 1. the [Sui source code](../build/install.md#source-code) installed in: `~/sui`

Reviewer: ```suggestion
1. the [Sui source code](../build/install.md#source-code) downloaded
```

I don't think installing in `~/sui` is a requirement, they just need to have the sui repo locally    @awelc to elaborate

Author: @sblackshear suggested this in the draft. I think we should do so in the central install docs so we might standardize across all docs. Or set a SUI_ROOT environment variable or some such.

Author: https://www.notion.so/Internal-Only-Sui-DevNet-Dogfooding-Guide-41d8c144897941f188d446f41411b919?d=926d0a283cae4272bf3bc91d47970960#dda2a8ffc0734f3793d13a356616564e

Author: Also, we can change this to a suggestion. WDYT?

Author: I made this a recommendation, along with the alternative environment variable

Reviewer: Sounds good

---
title: Experiment with Sui DevNet
---

Welcome to the beginnings of the Sui DevNet. It exists now to gain operational experience with the Sui software in a public setting. The Sui DevNet currently consists of:

* A four-validator network with all nodes operated by Mysten Labs. Clients send transactions and read requests via this endpoint: http://gateway.devnet.sui.io/
* A public network [Sui Explorer](https://github.com/MystenLabs/sui/tree/explorer-rest/explorer) for browsing the TestNet transaction history: http://explorer.devnet.sui.io
* A [Discord channel](https://discordapp.com/channels/916379725201563759/971488439931392130) for requesting test coins that can be used to pay for gas on the test network. These coins have no financial value and will disappear each time we reset the network.

> Many improvements to the Sui DevNet are underway, such as the ability to run FullNodes and use a browser-based wallet. See the Sui DevNet blog post announcement for full details on upcoming features.

Reviewer 2: Should we just say "full nodes" instead of "FullNodes"?

Author: Sure! Whatever works for you all. I got this from:
https://www.notion.so/mystenlabs/Fullnodes-in-the-Sui-Ecosystem-b28f6a2b0a6c4fdd86f1ea852ab49ac2

And tried to emulate:
https://aptos.dev/basics/basics-fullnodes/

Author: Done!

TODO: Create and link to Medium blog post.

## Purpose

> This procedure leverages the following components of the Sui blockchain:

Reviewer 2: I am not sure what "procedure" refers to here... How about something like: "DevNet deployment features the following component of the Sui blockchain"

Author: Let's hold on review until I get done with the doc. :-)

Author: Done!

- The Wallet CLI
    - creating and managing your private keys
    - submitting transactions for creating example NFTs on SUI
    - calling/publishing Move modules
- The Gas Faucet service
    - Transfer SUI tokens to you so that you can pay for the transactions
- SuiExplorer
    - View transactions and objects

TODO: Edit, format, and link from the text above to relevant pages once we agree to include it. No faucet docs will exist for DevNet per Chris.

## Prerequisites

### Set up environment

To use the Sui DevNet, you will need:

* the [`git` command line interface](https://git-scm.com/download/)
* coins requested through Discord (need channel link)

> * [Sui binaries in your PATH](https://docs.sui.io/build/install#binaries) environment variable

Reviewer 2: ```suggestion
* [Sui binaries](https://docs.sui.io/build/install#binaries) in your PATH environment variable
```

Author: > 

Done.

:tip: To reliably test DevNet with the latest Sui binaies, re-install them at least weekly.

> In addition, to conduct advanced work such as publishing a Move module or making a Move call, also download the Sui source code.

Reviewer 2: Should we add link to cloning instructions?

Author: Done!
